### DISCUSSION TITLE
Since version 9.0, getting negative electric use and no gas data in energy dashboard.

### DISCUSSION THREAD

**Discussion by MrLemur** [BottlecapDave/HomeAssistant-OctopusEnergy#553] [Q&A]
I am using the `current_consumption` sensors for both electric and gas. Since updating to v9.0, I get the following charts:
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/17055328/34fa568c-8ea9-4277-9eec-f00640708847)
For reference, this was from yesterday before the update:
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/17055328/92039f31-3e40-4888-8f9f-b9a70f1554e8)

Could this be due to the change to events in the new release?

**Comment by BottlecapDave** [comment]
This is not normal. That sensor is a bit of a legacy sensor and is a bit temperamental. The calculation logic hasn't changed in this release, but it did change to report the data via a different property based on HA recommendations. Are you seeing negative points if you look at the history of the sensor? Are you seeing any negativity with the current accumulative consumption sensor?

**Comment by MrLemur** [comment]
Current consumption history is not showing any negative values, and current accumulative sensor is showing as expected.
Current consumption for electric:
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/17055328/71596fc4-2f2d-4ec2-a34d-2ae80568134e)
Current accumulative for electric:
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/17055328/e213511c-b588-4cc3-b631-0631e5dc35a6)

Would it be advisable to move to the current accumulative sensor instead?

**Reply by deltamelter** [reply]
Seeing the same, no gas and weird negative and now all zero but not...
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/27061628/371e57aa-fc89-43d0-bb9e-38b65afd7bb7)
Using OHM and no negatives seen in actual sensors

**Reply by deltamelter** [reply]
also only since updgrade.  Note I had the cost set up differently than documented.  Was using `Use an entity with current price` with the current_rate entity but all worked fine until 9.0
last night changed to use `Use an entity tracking the total costs` to see if it made a difference, but doesn't seem to...

**Reply by BottlecapDave** [reply]
I'm using the total costs entity and it's still coming through fine for me. I also use the current accumulative consumption instead of the current consumption sensors. The current consumption sensor is a little legacy, but should still be working. I'll probably need to add some logging, as I'm wondering if the API is being a bit flakey.

**Reply by deltamelter** [reply]
I was using `current consumption` which has been fine until now, but just switched to `current accumulative consumption` and it looks fine now...

**Reply by deltamelter** [reply]
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/27061628/094f7c49-44bf-4ec6-9913-793874215fd4)

**Reply by MrLemur** [reply]
@deltamelter I've also changed to `current_accumulative` and it's showing correctly now, so happy with that.

**Reply by noisytwit** [reply]
I am using the same as you, current accumulative consumption as well as previous accumulative consumption, and both of those no longer work on the apex charts I had setup. I have checked the entities and I have no data being pulled on any of my previous days, but do have the current days accumulated data, it's just not being updated in the chart.

For context I only updated to v9 this evening. Prior to the update rhe charts worked. In addition if I click on the events, they are extremely slow to show the logs, and so I wonder if there is an API issue?

**Reply by ianmack-byte** [reply]
Me too, current_consumption started showing negative values, changed to current_accumulative and all OK
I cant see current_demand as an option for the Energy dashboard

**Reply by BottlecapDave** [reply]
@noisytwit  Clicking on events hit HA local data not the API, so if it's slow it's possibly something with your instance. If you're still having issues with the apex charts, have a look at the community docs as the values have changed.

@ianmack-byte Current demand has never been supported in the energy dashboard because the data metric is not supported.

**Comment by willenic** [comment]
> I'm using the total costs entity and it's still coming through fine for me. I also use the current accumulative consumption instead of the current consumption sensors. The current consumption sensor is a little legacy, but should still be working. I'll probably need to add some logging, as I'm wondering if the API is being a bit flakey.

Hi Dave, can I just check which sensor you are referring to here for total costs. I'm currently using ``` Electricity account ref Current Accumulative Cost ``` which seems to be working

**Reply by BottlecapDave** [reply]
@deltamelter said

> last night changed to use `Use an entity tracking the total costs` to see if it made a difference, but doesn't seem to

I was just stating that I use `Use an entity tracking the total costs` with the current accumulative cost entity with success, as per the docs.

**Comment by BottlecapDave** [comment]
I think I've found the issue and it looks like HA is not reading my updated last reset value. Im testing the solution at the moment.

**Comment by BottlecapDave** [comment]
This fix should now be available in the latest release (v9.0.2). It might take up to 24 hours for everything to stablise due to how the energy dashboard processes data.
